### PR TITLE
MANTA-4779 want node-fash to work without error on node v6

### PR DESCRIPTION
**Pull request by bahamas10**
This has been tested by testing joyent/buckets-mdplacement#12

From the `levelup` readme

> The level package is the recommended way to get started. It conveniently bundles levelup, leveldown and encoding-down. Its main export is levelup - i.e. you can do var db = require('level').

### CODE REVIEW

File: test/leveldb.test.js [hunk 1]
```diff
@@ -983,7 +983,7 @@ _testAllAlgorithms(function collision(algo, t) {
         pnodes: ['a', 'a'],
         vnodes: NUMBER_OF_VNODES,
         backend: fash.BACKEND.LEVEL_DB,
-        location: '/tmp/' + uuid.v4()
+        location: '/var/tmp/' + uuid.v4()
```
Comment (bahamas10) [inline test/leveldb.test.js]: I had some issues running out of memory on my zone when running the tests, and it seems that `/var/tmp` helped.  I couldn't get all the tests to work, but this helped the tests get farther without complaining about `ENOMEM`

File: package.json [hunk 2]
```diff
@@ -38,26 +38,24 @@
     "node": ">=0.8"
   },
   "dependencies": {
-    "assert-plus": "0.1.2",
-    "bignum": "0.6.0",
-    "bunyan": "*",
+    "assert-plus": "1.0.0",
+    "bignum": "0.13.0",
+    "bunyan": "2.0.2",
```
Comment (trentm) [inline package.json]: Latest bunyan 1.x please. 2.x is still "beta/experimental".

File: test/leveldb.test.js (hunk 1 above)
Comment (trentm) [reply test/leveldb.test.js]: Can we dig into the test failures to try to solve them? Or is that a science project?

File: package.json (hunk 2 above)
Comment (bahamas10) [reply package.json]: will do

File: test/leveldb.test.js (hunk 1 above)
Comment (bahamas10) [reply test/leveldb.test.js]: i'll check it out today and see how deep it goes

**Comment by bahamas10** [comment]
This latest commit makes this module pass `make check` and *most* of `make test`.

Tests
-------

This is a weird setup.  There are 3 test files:

- in_memory.test.js
- leveldb.test.js
- load.test.js

`in_memory` and `leveldb` seem to do the bulk of the testing, and they are both *working* perfectly against this branch.

`load`, however, doesn't work, and I'm convinced never worked.  It doesn't follow the normal flow of a `nodeunit` test, and really doesn't do anything except attempt to a load a leveldb directory that *does not* exist, and then start a restify server with no automated testing done against it.

First problem: even though `createIfMissing` is explicitly set to `true`, this value gets overridden silently to `false` automatically (??) in `fash.load` in the leveldb backend, with the comment:

> // don't create if loading from db

While I follow why this happens, this means that the test will always attempt to load a leveldb directory that was never created.  I've looked into the latest master on our fork and on the original node-fash, and this seems to always have been the case.... i.e.: why i think this never worked.

Also, I have no idea why restify is used.  Perhaps this file was used for manual testing, and then left in the tests directory accidentally? I don't know the history.  But the story remains - I don't think this test ever worked as a nodeunit test, and if we can remove it, we can also remove the restify dep.

File: lib/backend/leveldb.js [hunk 3]
```diff
@@ -763,7 +762,6 @@ ConsistentHash.prototype.addData = function addData(vnode, data, cb) {
 
                 return _cb(err);
             });
-            console.log('{"%s":"%s"}', vnode, data);
```
Comment (trentm) [inline lib/backend/leveldb.js]: This might be relevant output for `./bin/fash.js add-data`?

File: lib/backend/leveldb.js [hunk 4]
```diff
@@ -1017,7 +1015,6 @@ ConsistentHash.prototype.remapVnode = function remapVnode(newPnode, vnode, cb) {
                     return _cb();
                 }
             });
-            console.log('{"%s":"%s"}', newPnode, vnode);
```
Comment (trentm) [inline lib/backend/leveldb.js]: This might be relevant output for `./bin/fash.js remap-vnode`?  I'm not sure.

File: lib/backend/leveldb.js (hunk 3 above)
Comment (bahamas10) [reply lib/backend/leveldb.js]: agreed. will add back.

File: lib/backend/leveldb.js (hunk 4 above)
Comment (bahamas10) [reply lib/backend/leveldb.js]: agreed. will add back.

**Comment by bahamas10** [comment]
`make check` and `make test` run for this repo now.  I built this as a part of a build for https://github.com/joyent/buckets-mdplacement/pull/12 (https://jenkins.joyent.us/job/joyent-org/job/buckets-mdplacement/view/change-requests/job/PR-12/3/) and tested mdplacement on my lab machine and it all worked as expected as well.

File: package.json [hunk 5]
```diff
@@ -38,28 +38,25 @@
     "node": ">=0.8"
```
Comment (trentm) [inline package.json]: Let's consider bumping this to >=0.10 now.

And then we'll want a version bump here for release (could either be separate or part of this change). If we are bumping engines.node, we could bump the major version. Really no one uses 0.8 so it doesn't matter from that perspective. The changes in deps are a bigger change.  A new major ver is also a good time to change the "repository" field to point to joyent/node-fash.

**Review by trentm — DISMISSED** [review]

File: package.json (hunk 5 above)
Comment (bahamas10) [reply package.json]: agreed on all of this... i'll make these changes.

**Review by trentm — APPROVED** [review]
Could add a Jenkinsfile, here or separate ticket is totally fine.

**Comment by bahamas10** [comment]
https://jira.joyent.us/browse/MANTA-4781 for jenkins file

**Pull request merged**